src/libxtp/gwbse/gw.cc

```diff
@@ -251,111 +266,103 @@ Eigen::VectorXd GW::SolveQP(const Eigen::VectorXd& frequencies) const {
   return frequencies_new;
 }
 
-boost::optional<double> GW::SolveQP_Linearisation(double intercept0,
-                                                  double frequency0,
-                                                  Index gw_level) const {
+boost::optional<double> GW::SolveQP_Bisection(double freq_lb, double freq_ub,
```

Comment:
Bisection is guaranteed to converge so why return an optional?

Reply:
In this branch, bisection could fail if the bisection loop counter exceeds `_opt.g_sc_max_iterations`